src/main.c

```diff
@@ -223,14 +258,27 @@ int main(void)
     // clear in case we kept DFU_DBL_RESET_APP there
     (*dbl_reset_mem) = 0;
 
-    // start application
-    bootloader_app_start();
+    // register the app
+    uint32_t app_addr = bootloader_app_register();
+    if (!dfu_activate) {
```

Comment:
since we reset when `dfu_activate=1`, we can just skip the teardown and simply reset. Teardown is only needed when jumping to app. I will make changes accordingly.